### PR TITLE
Ch badges integrations

### PR DESCRIPTION
**Pull request by Byadiso**
### What does this PR do?
Add coverage badge, maintainability, and simple tests

### Description of Task to be completed?
Added coverage badge.
Setup of maintainability badges. 
Added simple tests with jest.

### How should this be manually tested?
step 1: clone git repo if not yet done.
step 2: cd Avengers-Phantom-FE.
step 3: yarn or npm install to install dependencies.
step 4: yarn or npm run test to test the codes.
step 5: Use google chrome for better rendering (google chrome redux).

### CODE REVIEW

File: src/Components/Homepage/MapSection.jsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ import Card from "@mui/material/Card";
 import Button from "@mui/material/Button";
 import Menu from "@mui/material/Menu";
 import { NavigationFormValidator } from "./NavigationFormValidator.jsx";
-import { MapContainer, TileLayer, useMap, Marker, Popup } from "react-leaflet";
+// import { MapContainer, TileLayer, useMap, Marker, Popup } from "react-leaflet";
```
Comment (Anguandia) [inline src/Components/Homepage/MapSection.jsx]: Are you sure this is not needed? please confirm with @Jeanndo and let me know

File: src/Components/dashboard/operatorsanddriver/DriverAndOperator.jsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ import Box from "@mui/material/Box";
 import Tab from "@mui/material/Tab";
 import TabContext from "@mui/lab/TabContext";
 import TabList from "@mui/lab/TabList";
-import Skeleton from "react-loading-skeleton";
+
```
Comment (Anguandia) [inline src/Components/dashboard/operatorsanddriver/DriverAndOperator.jsx]: and this too

**Review by Anguandia — CHANGES_REQUESTED** [review]
Good work @Byadiso, please address the few comments in addition to the in-code comments and let me know:
- there seem to be two branches to this chore, one being a misnomer while the other has the right name (spelling), which one, if we should, should we delete?
- the CI badge is missing, please consider setting up Github Actions on a separate PR/branch, preferably after this is merged

File: src/Components/Homepage/MapSection.jsx (hunk 1 above)
Comment (Byadiso) [reply src/Components/Homepage/MapSection.jsx]: -Thanks for the comment, I have uncommented the above line for import.

-The correct branch name will be "ch-badges-integrations" (the one with a PR), we will need to remove the other one!

File: src/Components/dashboard/operatorsanddriver/DriverAndOperator.jsx (hunk 2 above)
Comment (Byadiso) [reply src/Components/dashboard/operatorsanddriver/DriverAndOperator.jsx]: This one was not yet implemented, will take care during its implementation.

**Review by Anguandia — APPROVED** [review]

File: src/Components/dashboard/route/AddRouteInit.jsx [hunk 3]
```diff
@@ -8,148 +18,174 @@ import { Link } from "react-router-dom";
...
+        }
+        if (stop === "") {
+            setStopError(true);
+        } else {
+
+            console.log(routeCode, startLocation, endLocation, stop, name, "yessss");
+            const data = dispatch(saveRoute( { routeCode, startLocation, endLocation,name:'route', distance:'10km',duration:'1hr'}))
+            const error = localStorage.getItem("error");
+            if(error ){
+              console.log('error')
```
Comment (abraham-kwizera) [inline src/Components/dashboard/route/AddRouteInit.jsx]: As long as we do have a state that will help us to store error messages, hope these console log lines would be removed.

File: src/Components/dashboard/route/AddRoutes.jsx [hunk 4]
```diff
@@ -80,7 +80,9 @@ const AddRoutes = () => {
     if (stop === "") {
       setStopError(true);
     } else {
+
       console.log(routeID, startPoint, endPoint, stop);
```
Comment (abraham-kwizera) [inline src/Components/dashboard/route/AddRoutes.jsx]: Same here let's remove these console log lines

File: src/Components/dashboard/route/UpdateRouteButton.jsx [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+    const [startPointError, setStartPointError] = useState(false);
+    const [endPointError, setEndPointError] = useState(false);
+    const [stopError, setStopError] = useState(false);
+
+    const [openUpdater, setOpenUpdater] = useState(true);
+    const [open, setOpen] = useState(false);
+    const [updated, setupdated] = useState(false);
+
+    const handleClear = () => {
+        console.log("Content cleared");
```
Comment (abraham-kwizera) [inline src/Components/dashboard/route/UpdateRouteButton.jsx]: Most of your functions returns console logs messages in this file, hope to improve it better you may remove these lines too.

File: src/Components/dashboard/route/UpdateRouteButton.jsx [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+                                    error={stopError}
+                                />
+                                <AddIcon sx={{ py: 1, px: 1 }} onClick={handleAddRoute} />
+                            </Box>
+                            <Box>
+                                <List className="list-route">
+                                    <ListItem
+                                        sx={{ border: "1px solid #c4c4c4", width: "62%" }}
+                                    // secondaryAction={
+                                    //   // <IconButton edge="end" aria-label="delete">
```
Comment (abraham-kwizera) [inline src/Components/dashboard/route/UpdateRouteButton.jsx]: Let's remove these commented codes in this file, if they are not still needed

File: src/redux/Action/routes.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+  } catch (error) {
+    localStorage.setItem("error", error.message);
+    dispatch({ type: actions.ROUTE_DETAILS_FAIL, payload: error.message });
+  }
+};
+
+export const saveRoute = (route) => async (dispatch, getState) => {
+  try {
+    // const {
+    //   userSignin: { userInfo },
```
Comment (abraham-kwizera) [inline src/redux/Action/routes.js]: Let's improve our codes by removing these commented codes if they are not needed.

**Review by abraham-kwizera — COMMENTED** [review]
Hey, @Byadiso You did great, but I left some reviews to work on before we go. 

And remember to resolve these Conflicts listed in :point_right: [Conflicting Files](https://github.com/atlp-rwanda/Avengers-Phantom-FE/pull/42#:~:text=package%2Dlock.json,Reducer/index.js)

File: src/redux/Action/routes.js (hunk 7 above)
Comment (Byadiso) [reply src/redux/Action/routes.js]: Thank you for the comment @abraham-kwizera , I have resolved this , has been removed!

File: src/Components/dashboard/route/UpdateRouteButton.jsx (hunk 6 above)
Comment (Byadiso) [reply src/Components/dashboard/route/UpdateRouteButton.jsx]: Thank you for the comment @abraham-kwizera , I have resolved this

File: src/Components/dashboard/route/UpdateRouteButton.jsx (hunk 5 above)
Comment (Byadiso) [reply src/Components/dashboard/route/UpdateRouteButton.jsx]: Thank you for the comment @abraham-kwizera , I have resolved this

File: src/Components/dashboard/route/AddRoutes.jsx (hunk 4 above)
Comment (Byadiso) [reply src/Components/dashboard/route/AddRoutes.jsx]: Thank you for the comment @abraham-kwizera , I have resolved this

File: src/Components/dashboard/route/AddRouteInit.jsx (hunk 3 above)
Comment (Byadiso) [reply src/Components/dashboard/route/AddRouteInit.jsx]: Thank you for the comment @abraham-kwizera , I have resolved this

File: src/Components/dashboard/route/Skeletons/SkeletonElement.jsx [hunk 8]
```diff
@@ -1,4 +1,5 @@
 import React from 'react';
+<<<<<<< HEAD
 import Skeleton from'react-loading-skeleton';
 import {
```
Comment (mansurissa) [inline src/Components/dashboard/route/Skeletons/SkeletonElement.jsx]: Conflicts

**Review by mansurissa — CHANGES_REQUESTED** [review]
Resolve conflicts

File: src/Components/dashboard/route/Skeletons/SkeletonElement.jsx (hunk 8 above)
Comment (Byadiso) [reply src/Components/dashboard/route/Skeletons/SkeletonElement.jsx]: All conflicts has been resolved

**Pull request merged**